handbook/engineering.md

```diff
@@ -270,6 +247,127 @@ The following [Slack channels are maintained](https://fleetdm.com/handbook/compa
 | `#g-agent`                          | Zach Wasserman
 | `#_pov-environments`                | Ben Edwards
 
+## Scaling gotchas
+
+### Overall
+
+Nowadays, fleet, as a Go server, scales horizontally very well. It’s not very CPU or memory intensive. In terms of load in infrastructure, from highest to lowest is: MySQL, Redis, fleet.
+
+In general, if burning a bit of CPU or memory on the fleet side allows us to reduce the load on MySQL or Redis, we should do so.
+
+In a lot of cases, caching helps but given that we are not doing load balancing based on host id (i.e. make sure that the same host ends up in the same fleet server) this goes only so far. Caching host specific data is not done because round robin LB means all fleet instances end up circling the total list of hosts.
```

Comment:
```suggestion
In a lot of cases, caching helps but given that we are not doing load balancing based on host id (i.e. make sure that the same host ends up in the same fleet server) this goes only so far. Caching host specific data is not done because round robin LB means all Fleet instances end up circling the total list of hosts.
```